game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsFactoryTest.java

```diff
@@ -0,0 +1,137 @@
+  void getNamesWithTwoFiringGroups() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsFactory.builder()
+                .battleActions(mock(BattleActions.class))
+                .returnFire(MustFightBattle.ReturnFire.ALL)
+                .diceRoller((arg1, arg2) -> new DiceRoll())
+                .firingGroupSplitter(
+                    (arg1) ->
+                        List.of(
+                            FiringGroup.groupBySuicideOnHit(
+                                    "army", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0),
+                            FiringGroup.groupBySuicideOnHit(
+                                    "spies", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0)))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(
+        names,
+        is(
+            mergeSteps(
+                generalFightStepStrings(attacker, defender, "army"),
+                generalFightStepStrings(attacker, defender, "spies"))));
+  }
+
+  @Test
+  @DisplayName("With the default name UNITS, the step names should not include UNITS")
+  void getNamesWithDefaultUnitName() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsFactory.builder()
+                .battleActions(mock(BattleActions.class))
+                .returnFire(MustFightBattle.ReturnFire.ALL)
+                .diceRoller((arg1, arg2) -> new DiceRoll())
+                .firingGroupSplitter(
+                    (arg1) ->
+                        FiringGroup.groupBySuicideOnHit(
+                            UNITS, List.of(givenAnyUnit()), List.of(mock(Unit.class))))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(
+        "units should not be in the name",
```

Comment:
Why should units not be in the name?

I think this would be a stronger statement if it were something like:
"A battle with offense and all suicide units should be labelled as 'XYZ'"

With the above, I wonder why are we not checking for an exact name?

I suspect that would be more obvious if the checked values were not behind a method. I also notice there is logic for a magic value of emptry string in `generalFightStepStrings`. I think it might make for stronger tests to inline the function everywhere in tests to make the assertions explicit and clear.

Reply:
I had created the method in a separate PR for the purpose to reduce the size of the final PR.  It changes all of these step names in the same way and would make that PR very large.  Once that is in, I can inline the method in a later PR.

As for why "units" should not be in the name, that is the "human readable" subjective view.  The step name is either:
```
Attacker fires
Defender selects casualties
Defender removes casualties
```
vs 
```
Attacker units fire
Defender selects units casualties
Defender removes units casualties
```
I find the former looks better in this case.

But the map designer can give the group a name in the XML.  An example would be "AA", "Boomstick", or "AntiTankGun".  In that case, the name needs to be in the string:
```
Attacker AntiTankGun fire
Defender selects AntiTankGun casualties
Defender removes AntiTankGun casualties
```